src/Analyzer.Utilities/JsonLineNumberResolver.cs

```diff
@@ -115,20 +117,108 @@ public int ResolveLineNumber(string pathInExpandedTemplate)
             var originalTemplatePath = tokenFromOriginalTemplate.Path;
             var unmatchedPathInOriginalTemplate = pathInExpandedTemplate[originalTemplatePath.Length..].TrimStart('.');
 
+            bool pathsAreEqual = string.Equals(originalTemplatePath, pathInExpandedTemplate, StringComparison.OrdinalIgnoreCase);
+
+            bool unmatchedSegmentReferencesResourcesArray = unmatchedPathInOriginalTemplate.StartsWith("resources", StringComparison.OrdinalIgnoreCase);
+            bool unmatchedSegmentStartsWithIndexArray = unmatchedPathInOriginalTemplate.StartsWith("[", StringComparison.OrdinalIgnoreCase);
+            bool originalPathEndsWithResourcesArray = originalTemplatePath.EndsWith("resources", StringComparison.OrdinalIgnoreCase);
+
+            bool pathIsNotFromCopiedResource = !unmatchedSegmentReferencesResourcesArray
+                    && !unmatchedSegmentStartsWithIndexArray
+                    && !originalPathEndsWithResourcesArray;
```

Comment:
I don't think this is quite right.  For example, if the expanded path is *resources[0].dependsOn[3]* and the matched portion is *resources[0].dependsOn* (unmatched being *[3]*), `unmatchedSegmentStartsWithIndexArray` would be `true`, and it would continue and start looking for copied resources.  It should actually probably just return the line at *resources[0].dependsOn*.

Reply:
In this example, where does dependsOn[3] come from if it's not in the original template?

Reply:
Maybe variable evaluation?  Might be unlikely for dependsOn, but could happen with any array.  For a vnet/subnet, maybe *resources[0].properties.subnets[0]*.